src/engraving/dom/engravingitem.h

```diff
@@ -671,6 +671,7 @@ class EngravingItem : public EngravingObject
     virtual bool allowTimeAnchor() const { return false; }
 
     virtual bool hasVoiceApplicationProperties() const { return false; }
+    virtual bool discardDefaultVoiceApplication() const { return false; }
```

Comment:
It looks really weird... A virtual getter in one class and a non-virtual setter in another... And it looks like it's only needed in NotationInteraction

Reply:
It's because this check is only really needed for hairpins and dynamics (hence why only they have the setter), and the first parent class they have in common is EngravingItem. But I agree that it looks horrible, I'll find a better solution